test/singlenode_test.go

```diff
@@ -325,3 +327,57 @@ func TestSingleNodeCORS(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, "testhost.com", res.Header.Get("Access-Control-Allow-Origin"))
 }
+
+func TestSingleNodeJoinAfterClose(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+		return
+	}
+
+	_, finish := setupSingleNodeTest("TestJoinAfterClose")
+	defer finish()
+
+	scenarioJoinClosedRoom(t)
+}
+
+func TestAutoCreate(t *testing.T) {
+	disableAutoCreate := func(conf *config.Config) {
+		conf.Room.AutoCreate = false
+	}
+	t.Run("cannot join if room isn't created", func(t *testing.T) {
+		s := createSingleNodeServer(disableAutoCreate)
+		go func() {
+			if err := s.Start(); err != nil {
+				logger.Errorw("server returned error", err)
+			}
+		}()
+		defer s.Stop(true)
+
+		waitForServerToStart(s)
+
+		token := joinToken(testRoom, "start-before-create")
+		_, err := testclient.NewWebSocketConn(fmt.Sprintf("ws://localhost:%d", defaultServerPort), token, nil)
+		require.Error(t, err)
```

Comment:
Don't understand what this is testing.

Should this actually create a `testRTCClient` and check that it gets a 404?

Reply:
it's testing that it does not auto-create the room (and thus fail the join websocket connection)

currently the test client will panic when it fails. The websocket connection logic is the first bit of what createRTCClient is doing.